contractcourt/channel_arbitrator.go

```diff
@@ -492,15 +500,53 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 			nextState = StateCommitmentBroadcasted
 
 		// If this state advance was triggered by any of the
-		// commitments being confirmed, then we'll jump to the state
-		// where the contract has been closed.
+		// commitments being confirmed, then we'll jump to
+		// StateCommitmentConfirmed.
 		case localCloseTrigger, remoteCloseTrigger:
 			log.Infof("ChannelArbitrator(%v): state %v, "+
-				" going to StateContractClosed",
+				" going to SateCommitmentConfirmed",
 				c.cfg.ChanPoint, trigger)
+			nextState = StateCommitmentConfirmed
+		}
+
+	// If we are in this state it means a commitment has been seen
+	// on-chain, and we have written a CloseSummary to the log.
+	case StateCommitmentConfirmed:
```

Comment:
You're right. I think this could easily be solve by just logging the contract resolutions before marking the channel closed? I don't see a reason we need `nextState == StateContractClosed` before writing them. We could even write resolutions and the close summary in one atomic operation.

The reason I opted for introducing the state _before_ `StateContractClosed` was that it is a smaller alteration of the current flow (a channel is marked closed before starting incubation etc), and we don't have to make sure the actions taken in `StateContractClosed` are idempotent.

Let me know what you think!